README.md

```diff
@@ -1,2 +1,180 @@
-# ember-cli-workbox
-TBD
+ember-cli-workbox
+=================
+
+Service worker generator with precaching and some basic configurable options using [workbox-build](https://www.npmjs.com/package/workbox-build).
+
+For more details on Workbox check out:
+* [Workbox Google Developers](https://developers.google.com/web/tools/workbox/)
+* [Service Workers cookbook](https://serviceworke.rs/)
+
+Usage for Ember Cli
+-------------------
+
+`ember install ember-cli-workbox`
+
+### Configuration
+By default the service worker will be generated only for production builds and the service worker will be registered on an initializer.
+If you need to customize **ember-cli-workbox configuration** you can do it like this:
+
+```JavaScript
+//app/config/environment.js
+
+ENV['ember-cli-workbox'] = {
+  enabled: environment !== 'test',
+  debug
+};
+```
+* **enabled** - (Boolean) Addon is enabled. Default to true on production builds
+* **debug** - (Boolean) Log serviceworker states (registering, updating, etc)
+
+You can further customize ember-cli-workbox by setting **workbox configurations** in your environment.js file:
+
+```JavaScript
+//app/config/environment.js
+
+ENV['workbox'] = {
+  globPatterns: ['**\/*.{html,js,css}'],
+  globDirectory: './',
+  globIgnores: [],
+  ...
+};
+```
+
+* **swDest** - (String) The path to the final service worker file that will be created by the build process, relative to the build directory. Default path: './sw.js'
+* **globPatterns** - (Array of String) Files matching against any of these glob patterns will be included in the precache manifest. By default sw precaches all our ember application assets that match '**/*.{json,css,js,png,svg,eot,ttf,woff,jpg,gif,ico,xml,html,txt}'
+* **globDirectory** - (String) The base directory you wish to match globPatterns against, related to the build directory. Default  './'
+* **globIgnores** - (String or Array of String) Files matching against any of these glob patterns will be excluded from the file manifest, overriding any matches from globPatterns.
```

Comment:
Change syntax for Arrays?

(String or Array of String) => (String|String[])
(Array of Numbers) => (Number[])

Writing examples of each property can be really helpful (like runtimeCaching)

Reply:
i don't want to repeat workbox-build docs. I describe here some of the most important params but you have the links with complete and updated examples of workbox